src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/DuplicateWaysCheck.java

```diff
@@ -0,0 +1,115 @@
+                // Check to see that the edge has not already been seen
+                && !this.isFlagged(((Edge) object).getMasterEdgeIdentifier())
+                // Check to see that the edge is car navigable
+                && HighwayTag.isCarNavigableHighway(object)
+                // The edge is not part of an area
+                && !object.getTags().containsKey(AREA_KEY);
+    }
+
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Set<Segment> allEdgeSegments = new HashSet<>();
+        final Edge edge = (Edge) object;
+
+        final Rectangle bounds = edge.asPolyLine().bounds();
+        // Get Edges which are contained by or intersect the bounds, and then filter
+        // Out the non-master Edges as the bounds Edges are not guaranteed to be uni-directional
+        final Iterable<Edge> edgesInBounds = Iterables
+                .stream(edge.getAtlas().edgesIntersecting(bounds)).filter(Edge::isMasterEdge);
+
+        for (final Edge edgeInBounds : edgesInBounds)
+        {
+            // If the Edge found in the bounds has an area tag or if the Edge has a length of 0
+            // Or if the Edge has already been flagged before then continue because we don't want to
+            // Flag area Edges or duplicate Nodes
+            if (edgeInBounds.getTags().containsKey(AREA_KEY)
+                    || edgeInBounds.asPolyLine().length().equals(Distance.meters(ZERO_LENGTH))
+                    || this.isFlagged(edgeInBounds.getMasterEdgeIdentifier()))
+            {
+                continue;
+            }
+
+            // If the Set of Edges does not contain the Edge found in the bounds
+            if (!allEdgeSegments.containsAll(edgeInBounds.asPolyLine().segments()))
+            {
+                final List<Segment> edgeInBoundsSegments = edgeInBounds.asPolyLine().segments();
+
+                for (final Segment segment : edgeInBoundsSegments)
+                {
+                    // If a Segment in the Edge is found (check for partial duplication)
+                    if (allEdgeSegments.contains(segment))
+                    {
+                        this.markAsFlagged(edgeInBounds.getMasterEdgeIdentifier());
+                        return Optional.of(this.createFlag(edgeInBounds,
+                                this.getLocalizedInstruction(0, edgeInBounds.getOsmIdentifier())));
+                    }
+                }
+                // Add all Segments flattened to the Set of Segments
+                allEdgeSegments.addAll(edgeInBounds.asPolyLine().segments());
+            }
+            // A full duplicate Edge was found in the Set of allEdgePolyLines
+            else
+            {
+                this.markAsFlagged(edgeInBounds.getMasterEdgeIdentifier());
+                return Optional.of(this.createFlag(edgeInBounds,
+                        this.getLocalizedInstruction(0, edgeInBounds.getOsmIdentifier())));
```

Comment:
A full overlap has the same instruction. I'd suggest not worrying about the full overlap. Just flag whenever an overlap is found. Atlas has edges vs OSM has ways. Edges by definition are partial OSM ways, so full overlap or not, you'll almost always flag partial OSM ways anyways.